modules/lib/src/main/resources/assets/js/app/ui2/dialog/ConfirmationDialog.tsx

```diff
@@ -50,14 +52,110 @@ const ConfirmationDialogContent = forwardRef<HTMLDivElement, ConfirmationDialogC
     ({defaultSubmitEnabled = true, className, children, ...props}, ref): ReactElement => {
         return (
             <Dialog.Content ref={ref}
-                            className={cn('gap-3 min-w-170 max-w-fit', className)} {...props}>
+                            className={cn('text-main gap-3 min-w-170 max-w-fit', className)} {...props}>
                 <ConfirmationDialogProvider defaultSubmitEnabled={defaultSubmitEnabled}>{children}</ConfirmationDialogProvider>
             </Dialog.Content>
         );
     }
 );
 ConfirmationDialogContent.displayName = 'ConfirmationDialog.Content';
 
+//
+// * ConfirmationDialogGate
+//
```

Comment:
Move **Gate** into a separate component. We either don’t need to make it composite, or we don’t need a second level of nesting.
`ConfirmationDialog.Gate.Input` feels not right, so it should be either `ConfirmationDialog.Gate` or `Gate.*`.
I suggest going with the second option.